File: php/features.php
<?php
declare(strict_types=1);

// Bevwire SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class BevwireFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new BevwireBaseFeature();
            case "test":
                return new BevwireTestFeature();
            default:
                return new BevwireBaseFeature();
        }
    }
}

File: py/bevwire_sdk/features.py
# Bevwire SDK feature factory

from bevwire_sdk.feature.base_feature import BevwireBaseFeature
from bevwire_sdk.feature.test_feature import BevwireTestFeature


def _make_feature(name):
    features = {
        "base": lambda: BevwireBaseFeature(),
        "test": lambda: BevwireTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
